vm-aws-yaml/Pulumi.yaml

```
@@ -0,0 +1,124 @@
+        default: 10.0.0.0/16
+
+resources:  
+    # Create VPC
+    vpc:
+        type: aws:ec2:Vpc
+        properties:
+            cidrBlock: 10.0.0.0/16
+            enableDnsHostnames: true
+            enableDnsSupport: true 
+    # Create an internet gateway
+    gateway:
+        type: aws:ec2:InternetGateway
+        properties:
+            vpcId: ${vpc.id}
+            tags:
+                Name: vm-aws-yaml-ig
+    # Create a subnet that automatically assigns new instances a public IP address.
+    subnet:
+        type: aws:ec2:Subnet
+        properties:
+            vpcId: ${vpc.id}
+            cidrBlock: 10.0.1.0/24
+            mapPublicIpOnLaunch: true
+            tags:
+                Name: vm-aws-yaml-subnet
+    # Create a route table
+    routeTable:
+        type: aws:ec2:RouteTable
+        properties:
+            vpcId: ${vpc.id}
+            routes:
+                - cidrBlock: 0.0.0.0/0
+                  gatewayId: ${gateway.id}
+            tags:
+                Name: vm-aws-yaml-routetable
+    # Associate the route table with the public subnet.
+    routeTableAssociation:
+        type: aws:ec2:RouteTableAssociation
+        properties:
+            subnetId: ${subnet.id}
+            routeTableId: ${routeTable.id}
+    # Create EC2 Security Group
+    secGroup:
+        type: aws:ec2:SecurityGroup
+        properties:
+            description: Enable HTTP access
+            vpcId: ${vpc.id}
+            ingress:
+                - fromPort: 80
+                  toPort: 80
+                  protocol: tcp
+                  cidrBlocks: 
+                    - 0.0.0.0/0
+            egress:
+                - fromPort: 0
+                  toPort: 0
+                  protocol: -1
+                  cidrBlocks:
+                    - 0.0.0.0/0
+            tags:
+                Name: vm-aws-yaml-secgrp
+    # Create EC2 server
+    server:
+        type: aws:ec2:Instance
+        properties:
+            instanceType: ${instanceType}
+            subnetId: ${subnet.id}
+            vpcSecurityGroupIds: 
+                - ${secGroup}
+            userData: ${userData}
+            ami: ${ami}
+            tags:
+                Name: vm-aws-yaml-webserver
+
+variables:
+    # Create EC2 AMI
+    ami:
+        fn::invoke:
+            function: aws:ec2:getAmi
+            arguments:
+                filters:
+                    - name: name
+                      values: 
+                        - "amzn-ami-hvm-*"
+                owners: 
+                    - "137112412989"
+                mostRecent: true
+            return: id    
+    # User data to start a HTTP server in the EC2 instance
+    userData: "#!/bin/bash
+            
+            echo \"Hello, World from Pulumi!\" > index.html
+            
+            nohup python -m SimpleHTTPServer 80 &"
+            
+
+outputs:
+    # Export the public IP and DNS
+    publicIP: ${server.publicIp}
+    publicDNS: ${server.publicDns}
```

I've generally been calling these exports just `ip` and `hostname`. Ok with those, just for consistency?

```suggestion
    ip: ${server.publicIp}
    hostname: ${server.publicDns}
    url: http://${${server.publicDns}}
```

Adding `url` is also nice because you can immediately click it to open a browser, and on the template pages, we can do things like we do here, with `open $(pulumi stack output url)`, for a bit more engagement: https://www.pulumi.com/templates/container-service/aws/